hellodagger/src/main/java/com/airbnb/mvrx/helloDagger/base/BaseActivity.kt

```diff
@@ -0,0 +1,26 @@
+package com.airbnb.mvrx.helloDagger.base
+
+import android.os.Bundle
+import androidx.annotation.LayoutRes
+import androidx.appcompat.app.AppCompatActivity
+import com.airbnb.mvrx.helloDagger.appComponent
+import com.airbnb.mvrx.helloDagger.di.AssistedViewModelFactory
+import javax.inject.Inject
+
+/**
+ * Base class for Activities.
+ *
+ * This class contains an injected map of [AssistedViewModelFactory]s to make it easier to create
+ * and instantiate requested ViewModels.
+ */
+abstract class BaseActivity(@LayoutRes layoutResId: Int = 0) : AppCompatActivity(layoutResId) {
+
+    @Inject
+    lateinit var viewModelFactoryMap: @JvmSuppressWildcards Map<Class<out BaseViewModel<*>>, AssistedViewModelFactory<*, *>>
```

Comment:
Yep. It would be simpler to simply add a provision for this dependency in the AppComponent.
I honestly don't have a better reason for why I injected in the activity other than "I've always done it this way"

Will change to dependency provision in the AppComponent and update the PR soon.